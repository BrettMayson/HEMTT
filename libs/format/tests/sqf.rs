use hemtt_format::{FormatterConfig, format_sqf};

const ROOT: &str = "tests/sqf/";

macro_rules! sqf {
    ($dir:ident) => {
        paste::paste! {
            #[test]
            fn [<sqf_ $dir>]() {
                insta::assert_snapshot!(sqf(stringify!($dir)));
            }
        }
    };
}

sqf!(arrays);
sqf!(blocks);
sqf!(cba);
sqf!(comments);
sqf!(format);
sqf!(hash);
sqf!(numbers);
sqf!(if);
sqf!(macros);
sqf!(preserve_lines);
sqf!(private);
sqf!(sameline);

fn sqf(file: &str) -> String {
    format_sqf(
        &std::fs::read_to_string(format!("{ROOT}{file}.sqf")).expect("Failed to read test file"),
        &FormatterConfig::default(),
    )
    .unwrap_or_else(|err| format!("ERROR: {err}"))
}
