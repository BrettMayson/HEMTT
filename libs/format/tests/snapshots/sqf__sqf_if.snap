---
source: libs/format/tests/sqf.rs
expression: sqf(stringify! (if))
---
if (true) then {
    _x = 5;
    for "_i" from 0 to 10 do { _sum = _sum + _i; };
}; // test

if !(call _test) then { systemChat "true branch"; };

if ((abs (_actualOffset - _configOffset)) > 0.1) then {};
if ((abs (_actualOffset - _configOffset)) > 0.1) then {};

if (_xPosOptic isEqualTo [0, 0, 0]) exitWith { -999 }; // e.g. akm has no proxy
